### PR TITLE
AB#6705 Add functionality to website that parses web requests with special querystring information

### PR DESCRIPTION
**Pull request by chpa7**

### CODE REVIEW

**Comment by scottmulligan** [comment]
**Page Redirection**

When using "router.push", it looks like the initial page loads before evaluating the redirect to the Session page. I don't know much about redirects in next.js but the page flickering is not something I would want to see as an end user.

Are there any alternatives or ways to evaluate the logic before the home page loads? If not then maybe we can look at updating the QR code to direct the user to an empty "utility" page that would be responsible for the redirection. Open to ideas on this one.

**CDP Data Collection**

I might need some input from @jflheureux on this one. I'm not 100% sure what we would want to have listed in the guest profile for visitors coming through a QR code scan.

After coming to the website through a QR scan link, I see 4 MOBILE_WEB sessions listed in my guest profile. There is something weird going on with incoming guests.

cdp_qr_1

And in the last session, the "QR Code TV Scan" is listed twice. This might be a more broad problem (maybe we have duplicate events in other cases as well). If the duplicate event issue is more broad then we can log it as a separate bug to be worked on.

cdp_qr_2

**Review by scottmulligan — APPROVED** [review]
Looks good! Final deployment test before merging.

File: Website/src/rendering/src/pages/_app.tsx [hunk 1]
```diff
@@ -41,6 +47,29 @@ function App({ Component, pageProps, router }: AppProps): JSX.Element {
...
+      // Finally redirect to the specific session page on the website if one is found
+      // otherwise redirect to the sessions page
+      (async () => {
+        await closeCurrentSession();
+        await logQRCodeEvent('QR Code TV Scan');
+
+        const sessionURL = await getSessionURLByContentHubID(contentHubSessionId);
+        if (sessionURL) {
+          const urlParts = sessionURL.split('/');
+          const websiteSessionURL = `/sessions/${urlParts[urlParts.length - 1]}`;
```
[the remark this answers is not in the document]
Comment (chpa7) [reply Website/src/rendering/src/pages/_app.tsx]: You are right Scott, much simpler this way! Updating

[the remark this answers is not in the document]
Comment (chpa7) [reply Website/src/rendering/src/pages/_app.tsx]: New PR located here https://github.com/Sitecore/Sitecore.Demo.Edge/pull/470

**Pull request merged**